pkg/api/v1/service/util.go

```diff
@@ -70,7 +70,8 @@ func GetLoadBalancerSourceRanges(service *v1.Service) (utilnet.IPNetSet, error)
 // ExternalPolicyLocal checks if service has ETP = Local.
 func ExternalPolicyLocal(service *v1.Service) bool {
 	if service.Spec.Type != v1.ServiceTypeLoadBalancer &&
-		service.Spec.Type != v1.ServiceTypeNodePort {
+		service.Spec.Type != v1.ServiceTypeNodePort &&
+		(service.Spec.Type != v1.ServiceTypeClusterIP || len(service.Spec.ExternalIPs) == 0) {
```

Comment:
You don't need to check `ClusterIP` specifically; presumably we wany any type of service that is allowed to have `ExternalIPs` to be able to have `ExternalTrafficPolicy`.